### PR TITLE
Add ability to run feature tests for pr

### PR DESCRIPTION
**Pull request by TravisEz13**

### CODE REVIEW

File: docs/testing-guidelines/testing-guidelines.md [hunk 1]
```diff
@@ -76,6 +76,15 @@ Additionally, the tag:
 
 * `SLOW` indicates that the test takes a somewhat longer time to execute (97% of our `CI` tests take 100ms or less), a test which takes longer than 1 second should be considered as a candidate for being tagged `Slow`
 
+#### Requesting additional tests for a PR
+
+In our CI systems, normally only tests tagged with `CI`.  If in the first line of the last (most recent) commit description you add `[Feature]`,
```
Comment (JamesWTruher) [inline docs/testing-guidelines/testing-guidelines.md]: ... we normally run only tests ...

File: docs/testing-guidelines/testing-guidelines.md [hunk 2]
```diff
@@ -76,6 +76,15 @@ Additionally, the tag:
...
+we will ensure that we will also run the tests tagged with `Feature`.  When you would want to do this:
+
+- You have added or changed a `Feature` test.
+- A maintainer asks you to run the `Feature` tests.
+- Based on pasted experience, you are confident that a maintainer will ask you to run the `Feature` tests.
```
Comment (JamesWTruher) [inline docs/testing-guidelines/testing-guidelines.md]: passed

File: tools/appveyor.psm1 [hunk 3]
```diff
@@ -85,6 +85,13 @@ Function Test-DailyBuild
     {
         return $true
     }
+    
+    # if [Feature] is in the commit message,
+    # Run Daily tests
+    if($env:APPVEYOR_REPO_COMMIT_MESSAGE -match '(\[(?i)feature\]){1}')
```
Comment (JamesWTruher) [inline tools/appveyor.psm1]: i think this need only be:
`-match '\[feature\]'`
you don't seem to be doing anything with the match, also our `-match` is case insensitive by default

File: tools/appveyor.psm1 [hunk 4]
```diff
@@ -174,11 +181,14 @@ function Invoke-AppVeyorInstall
 {
     if(Test-DailyBuild){
         $buildName = "[Daily]"
-        if($env:APPVEYOR_PULL_REQUEST_TITLE)
+
+        # Add daily to title if it's not already there
+        # It can be there already for rerun requests
+        if($env:APPVEYOR_PULL_REQUEST_TITLE -and $env:APPVEYOR_PULL_REQUEST_TITLE  -notmatch '^\[Daily\].*$')
```
Comment (JamesWTruher) [inline tools/appveyor.psm1]: the `.*$` is extraneous. Also, if you have the following pull request 
`fix things in [daily]` you would get `[Daily]fix things in [daily]` is that ok?

File: tools/appveyor.psm1 [hunk 5]
```diff
@@ -174,11 +181,14 @@ function Invoke-AppVeyorInstall
...
         {
             $buildName += $env:APPVEYOR_PULL_REQUEST_TITLE
         }
-        else
+        elseif($env:APPVEYOR_REPO_COMMIT_MESSAGE  -notmatch '^\[Daily\].*$'  )
```
Comment (JamesWTruher) [inline tools/appveyor.psm1]: is the tag only when `[daily]` is at the front of the commit message or anywhere? Also, is this case sensitive? if so, then `-notcmatch` should be used

File: tools/travis.ps1 [hunk 6]
```diff
@@ -96,7 +96,9 @@ function Set-DailyBuildBadge
 # TRAVIS_EVENT_TYPE: Indicates how the build was triggered.
 # One of push, pull_request, api, cron.
 $isPR = $env:TRAVIS_EVENT_TYPE -eq 'pull_request'
-$isFullBuild = $env:TRAVIS_EVENT_TYPE -eq 'cron' -or $env:TRAVIS_EVENT_TYPE -eq 'api'
+
+# Run a full build if the build was trigger via cron, api or the commit message contains `[Feature]`
+$isFullBuild = $env:TRAVIS_EVENT_TYPE -eq 'cron' -or $env:TRAVIS_EVENT_TYPE -eq 'api' -or $env:TRAVIS_COMMIT_MESSAGE -match '(\[(?i)feature\]){1}'
```
Comment (JamesWTruher) [inline tools/travis.ps1]: i don't understand the capture, it seems unused. would `-match '\[feature\]'` suffice?

File: tools/appveyor.psm1 (hunk 5 above)
Comment (TravisEz13) [inline tools/appveyor.psm1]: Only when it's at the front, preventing this code from running twice (In the case of a rerun through the AppVeyor UI)

File: tools/appveyor.psm1 (hunk 4 above)
Comment (TravisEz13) [inline tools/appveyor.psm1]: Think that's ok... this is supposed to put [Daily] at the front... if it's not at the front, add it.

**Review by JamesWTruher — APPROVED** [review]

**Pull request merged**